src/main/java/core/basesyntax/strategy/TransactionProcessorServiceStrategy.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitDb;
+import core.basesyntax.model.Transaction.Operation;
+import core.basesyntax.service.TransactionProcessorService;
+import core.basesyntax.service.impl.IncomingTransactionProcessorService;
+import core.basesyntax.service.impl.OutgoingTransactionProcessorService;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TransactionProcessorServiceStrategy {
+    private final Map<Operation, TransactionProcessorService> map = new HashMap<>();
+
+    public TransactionProcessorServiceStrategy(FruitDb db) {
+        TransactionProcessorService incoming = new IncomingTransactionProcessorService(db);
+
+        map.put(Operation.BALANCE, incoming);
```

Comment:
balance is revision operation, it should replace current quantity, not be added to it